plugins/categoryaslink/class.categoryaslink.plugin.php

```diff
@@ -116,6 +116,15 @@ function categoryUrl($category, $page = '', $withDomain = true) {
         }
         $category = (array)$category;
 
+        // If SubcommunitiesPlugin is enabled and the category use RedirectUrl feature.
+        if (class_exists('SubcommunitiesPlugin') && $category['RedirectUrl']) {
+            $requestRoot = Gdn::request()->getRoot();
+            // Remove $requestRoot in $destination because it will appear twice because of url() or safeURL().
+            $destination = safeURL($category['RedirectUrl']);
+            return $requestRoot && strpos($destination, $requestRoot) === 0 ?
+                substr($destination, strlen($requestRoot)) : $destination;
+        }
```

Comment:
This code works, there's nothing wrong with it per se, but you're checking for `$category['RedirectUrl']` twice. If you checked for `$category['RedirectUrl']` first and then inside that block check for `class_exists('SubcommunitiesPlugin')` and handle both scenarios with an `else` statement this baby would really shine.